src/T2DMap.cpp

```diff
@@ -215,6 +215,21 @@ QColor T2DMap::getColor(int id)
         break;
     default: //user defined room color
         if (!mpMap->customEnvColors.contains(env)) {
+            if (16 < env < 232)
```

Comment:
Not valid C/C++ code - needs to be `if (env > 15 && env < 232)`.

We need to change to have (a new) `case 0:` as `mpHost->mBlack_2`, `case 8:` as `mpHost->mLightBlack_2` and remove `case 16:` so that the colour codes match up to the correct **ANSI** values (0 is Black, 7 is Light Grey (`mpHost->mWhite_2`) 8 is Dark Grey (`mpHost->mLightBlack_2`), 15 is White (`mpHost->mLightWhite_2`) and 16 is the **first** of the 6x6x6 RxGxB colour range which does *happen* to be black as well (r=0, g=0, b=0) but should not be one that the user can override from the `Mapper colors` tab of the preferences.

Reply:
Ah, that'd be the issue with the grayscale. Lovely.

Reply:
Changed.